lfs/http.go

```diff
@@ -118,8 +119,11 @@ func (c *Configuration) HttpClient() *HttpClient {
 	}
 
 	sslVerify, _ := c.GitConfig("http.sslverify")
+	tr.TLSClientConfig = &tls.Config{}
 	if sslVerify == "false" || Config.GetenvBool("GIT_SSL_NO_VERIFY", false) {
-		tr.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		tr.TLSClientConfig.InsecureSkipVerify = true
+	} else {
+		tr.TLSClientConfig.RootCAs = getRootCAsForHost(host)
```

Comment:
Doh, much better idea. It already is cached, meaning this PR breaks it since we'll want different http client objects per host. That `c.httpClient` needs to become a `map[string]*HttpClient`.